src/holofuel/components/RecentCounterparties/RecentCounterparties.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import { isEmpty } from 'lodash/fp'
+import cx from 'classnames'
+import HashIcon from 'holofuel/components/HashIcon'
+import { presentAgentId } from 'utils'
+import './RecentCounterparties.module.css'
+
+export default function RecentCounterparties ({ agents, selectedAgentId, selectAgent, className }) {
+  if (isEmpty(agents)) return null
```

Comment:
I've never figured out how I want to handle this. Should the component be responsible for deciding whether to display itself, or should this logic be in the parent?

I go back and forth, and it might be that there isn't a single answer and it varies depending on the situation.

Curious your thoughts on this.

Reply:
Sensible default logic in component with parent access to override methods?

Reply:
you guys are talking about something more complicated than I meant.

I'm just wondering about having this code in the parent component

`{!isEmpty(agents) && <RecentCounterparties />}`

vs putting it in the component itself

`if (isEmpty(agents)) return null`

@JettTech your question about having the header visible even if there are no agents to list is good one, specific to this component, and a UX question we should throw back to C&Z